### PR TITLE
Do hydra to swagger specs migration for azure-sdk-for-net

### PR DESCRIPTION
**Pull request by Monicalyy**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@Monicalyy can you provide swagger spec PR link.

**Comment by darshanhs90** [comment]
can you update https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Authorization/Management.Authorization/Properties/AssemblyInfo.cs with the new version of 2.6.0

**Comment by Monicalyy** [comment]
@shahabhijeet  Because we compared Hydra and swagger files and found out that we don't need to modify the swagger Json. So we don't have any PR about Swagger.
 About generate.cmd, It seems like we don't need to update the specFile, Maybe we need to update the autoRestVersion to the latest version (1.2.0-Nightly20170714) right?

**Comment by shahabhijeet** [comment]
@Monicalyy what I am clear is:
1) why are generating? as I see you are increasing the version number
2) If you are regenerating, where is the update for the swagger spec?
That is the reason I am asking if you can provide what commit of swagger spec are you generating for?

Also provide a detailed description as to what is the reason for this PR.

**Comment by Monicalyy** [comment]
@shahabhijeet 
Because we received a task to generating Swagger specs from Hydra specs for authorization. 
Document says the spec might already be generated, just validate it covers all the API and filters. So After compared Hydra and swagger, We found we don't need to modify the swagger spec. But After generated the swagger specs there has some changes for Authorization project. So we did the test for the changes and created this pull request.
Maybe I did an unintelligible description, So if you have more question, could you contact Haridarshan Hosakere Sudhakar <hahosake@microsoft.com> online?

**Comment by darshanhs90** [comment]
@shahabhijeet Maybe there was some confusion regarding updating the assembly file version.
we were under the impression that we need to update the version for every change done to the sdk and publish[rather it be from swagger specs or sdk repo alone]
Since this PR doesnt involve changes to the swagger specs,looks like we dont need to update the version.
Hope that clarifies your queries.

File: src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj [hunk 1]
```diff
@@ -20,6 +20,9 @@
   </ItemGroup>
 
   <ItemGroup>
+    <None Update="SessionRecords\Authorization.Tests.BasicTests\ProviderOperationsMetadataListGetTests.json">
```
Comment (shahabhijeet) [inline src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj]: @Monicalyy please replace all these lines with something like below.
This helps in maintaining project files concise.
< ItemGroup>
    < None Update="SessionRecords \ ** \ *.json">
      < CopyToOutputDirectory>PreserveNewest< /CopyToOutputDirectory>
    < /None>
< /ItemGroup>

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@Monicalyy once the minor update is made, this PR is ready to be merged.

File: src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj (hunk 1 above)
Comment (Monicalyy) [inline src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj]: @shahabhijeet  I have finished the change. Please review.

**Comment by darshanhs90** [comment]
:shipit:

**Comment by shahabhijeet** [comment]
@azuresdkci  test this please

File: src/SDKs/Authorization/Authorization.Tests/Tests/BasicTests.cs [hunk 2]
```diff
@@ -929,6 +930,40 @@ public void RoleDefinitionCreateTests()
...
+                }
+
+                var providerOperationsMetadata = client.ProviderOperationsMetadata.Get("Microsoft.Web", API_VERSION);
+                Assert.NotNull(providerOperationsMetadata);
+                Assert.NotNull(providerOperationsMetadata.DisplayName);
+                Assert.NotNull(providerOperationsMetadata.Id);
+                Assert.NotNull(providerOperationsMetadata.Name);
+                Assert.NotNull(providerOperationsMetadata.Operations);
+                Assert.NotNull(providerOperationsMetadata.ResourceTypes);
+                Assert.NotNull(providerOperationsMetadata.Type);
```
Comment (darshanhs90) [inline src/SDKs/Authorization/Authorization.Tests/Tests/BasicTests.cs]: can you add a negative test for this get method?

**Comment by shahabhijeet** [comment]
@darshanhs90 can you use the code review tool to request for change. This way it shows red cross next to your name. Once you approve only then I will merge.

**Comment by darshanhs90** [comment]
@Monicalyy could you update the tests,so that we can finish this PR

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
once the CI passes this is good to merge

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@Monicalyy Authorization tests have failed.
I had disabled all Authorization tests so far, so today I enabled them back and they started failing.

**Comment by darshanhs90** [comment]
@shahabhijeet The tests are passing in both record and playback mode on our local system.
It is failing with the following issue,**Subscription Id is not present in the recorded mock**
even though it is part of the recorded mock.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**